src/dict.c

```diff
@@ -702,7 +702,7 @@ unsigned int dictGetSomeKeys(dict *d, dictEntry **des, unsigned int count) {
     unsigned long maxsteps;
 
     if (dictSize(d) < count) count = dictSize(d);
-    maxsteps = count*10;
+    maxsteps = (unsigned long)count*10;
```

Comment:
Can this actually overflow? I thought C type promotions happened before the operations, so count would become an unsigned long before the multiplication.

Reply:
I was also skeptical, and write a test.
```c
    unsigned int i = INT_MAX;
    long long j = i * 10;
    printf("%u, %lld\n", i, j);
```
Result: 2147483647, 4294967286.
Because in `count*10`, count and 10 are both int, so count is not converted to unsigned int.
`count*10` can modify to `(unsigned long)count * 10` or `count * 10LL` will be ok.

Reply:
Yeah, re-reading through all of the promotion rules, your proposal is the recommended approach.